### PR TITLE
Add descriptions from the README.md file's table to appear in TrollApps apps.json

### PR DESCRIPTION
**Pull request by jfalus**
Updates the apps.json generation script to add the app's description to the TrollApps listing from the README.

Also adds the Last Updated Date to the end of the description, and the Source/Maintainer column as the developer. Note: Source/Maintainer sometimes has the developer, and sometimes has just sources. I decided to add it though since it often has the developer.

Implementation Notes:

App Names in the README are matched with App Names from the releases by removing all white space and the making every letter lowercase. This works for about 80% of cases, but some have names with different characters. For example: MeituPlusPlus vs Meitu++.

This searches the second table in the README (the first one being the Known Reposters). So, if a table is added before this, the script won't work. The script will also not work if the table headers are changed in the README, but not in the script.

### CODE REVIEW

**Comment by ArianaMinajj** [comment]
Hello,

Would improving the matching between App Names in the README and releases help by removing things such as spaces, special characters, and any duplicates? Can this increase the success rate of the script?

Additionally, do you think this could potentially make it compatible again with other signing platforms such as ESign or SideStore?

**Comment by jfalus** [comment]
Hi

Things that would improve the success of this script would be if the "App Name" column in the README's table had the same names as the app names from the bundled.csv. My code can account for spaces and different capitalization, but not extra symbols.

Esign uses an entirely different structure for their sources files. I looked at Esign's built in source (https://esign.yyyue.xyz/app.json), and it doesn't have the "versions" list for each app. Since the ESign source is structured differently from the Trollstore source, they won't be able to be used in both apps.

SideStore seems to require more to set up than just installing it, so I haven't installed it yet. I've tried to match this (https://wiki.sidestore.io/advanced/app-sources/) as well as possible, but may have messed it up. Let me know if the current version works with it.

Finally, I added Status to the description. The README file's table seems to use GitHub emoticons. When read by the program, these will show up as :emoji:. Example: 💡 will appear as :bulb:. If you don't want this, I'd recommend changing from the GitHub emoticons to emojis copied from a website like https://emojihub.org.

**Comment by ArianaMinajj** [comment]
Thank you for your detailed feedback. I've spent a good amount of time understanding how your script works, and I've already begun the process of reworking the app names for each release even if this is still a time-consuming task.

With regards to compatibility with both TrollApps and ESign/GBox, since there's no solution, I plan on asking someone to maintain an updated fork specifically for that purpose.

As for SideStore, I will look into it as soon as possible! I appreciate your efforts to match their setup process and will let you know how it works with the current version.

Finally, in response to your point about the Status in the description, I'll consider switching from these to emojis copied from that website.

I'll keep you updated!

**Comment by ArianaMinajj** [comment]
I'm ready to merge this but it's indicating me that there are some conflicts that need to be resolved. Any idea on how to do that?

**Comment by jfalus** [comment]
Hi,

TL;DR: Click the "Resolve conflicts" button, ignore apps.json, make sure bundleId.csv is up to date, and then merge.

Apps.json will always be different (since mine has the descriptions while yours has new apps). Completing the merge request should regenerate it (as well as the bundleId.csv).

But, based on the code, I'd make sure the bundleId.csv does match the current version of bundleId.csv in your repository (can copy the file's contents on your end and paste it into the editor when clicking "resolve conflicts")

I can't click Resolve conflicts (since I don't have write access to the repository). But, let me know if you have any questions.

P.S. If it helps, here is a page from GitHub explaining how to resolve the conflicts: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github

**Comment by ArianaMinajj** [comment]
Everything went smoothly. A few apps require some changes but this is overall working flawlessly.

Thank you so much for your contribution, this is a huge change for the repository!
